src/services/accounts/AccountsValidateService.ts

```diff
@@ -26,16 +32,28 @@ export class AccountsValidateService extends AbstractService {
 
 		if (defaults.allowedEncodedLengths.includes(u8Address.length)) {
 			const [isValid, , , ss58Prefix] = checkAddressChecksum(u8Address);
-
+			const networkName = allNetworks.filter(
+				(network) => network.prefix === ss58Prefix
+			)[0].network;
+			if (isHex(address)) {
+				accountId = address;
+			} else {
+				const registry = new TypeRegistry();
```

Comment:
To handle the registry correctly we can leverage the decorated api. See [ValidateAddressController](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/controllers/accounts/AccountsValidateController.ts#L9). The controller passes an api down to the service. Therefore you can use `this.api.registry`. The reason we dont want to use `new TypeRegistry` is because it doesn't have any metadata attached to it specific to the connected chain. Instead it probably has static substrate metadata.